tagController.js

```diff
@@ -70,38 +80,70 @@ function tagAddRow(){
 function tagRemoveRowById(tagRowId){
     var tagTable = document.getElementById('tagTable');
     var tagRow = document.getElementById('tagRow_' + tagRowId);
-    var tagRowChecked = document.getElementById('tagRow_' + tagRowId + "_checkBox");
-    var tagCost = document.getElementById("tagRow_" + tagRowId + "_costBox");
-    let tagRowSelect = tagRow.children[1].children[0];  //yes, this assumes a hardcoded order of child elements ('bad form' - Hook)
-    let tagRowSelectValue = tagRowSelect.value.toString();
-
-    //remove tag from the tagCost if tagRow is checked
-    if(tagRowChecked.checked == 1){
-        totalTagSum(-tagCost.innerText);
+    let tagHiddenId = document.getElementById("tagRow_" + tagRowId + "_tagId");
+    let tagRowCheck = document.getElementById("tagRow_" + tagRowId + "_checkBox");
+
+    //removes tagId from Unit Data's tags[]
+    let tagCounter;
+    let removeIdIndex = -1;
+    for(tagCounter = 0; tagCounter < mainUnitData.tags.length; tagCounter++){
+        let tagData = mainUnitData.tags[tagCounter];
+        if(tagData.id == tagHiddenId.value){
+            removeIdIndex = tagCounter;
+        }
     }
-    
-    //removes tagId from Unit Data's tagIds[]
-    let removeIdIndex = mainUnitData.tagIds.indexOf(tagRowSelectValue, 0);
-    mainUnitData.tagIds.splice(removeIdIndex, 1);
-    
-    tagTable.deleteRow(tagRow.rowIndex);
 
+    if(tagRowCheck.checked == 1){
+        tagRowCheck.checked = 0;
+        tagRowChangeActive(tagRowId);
+    }
+    
+    if(removeIdIndex != -1){
+        mainUnitData.tags.splice(removeIdIndex, 1);
+    }
     //update other take row selects with the change
     tagListsUpdate();
 
-    //TODO - adjust total tag costs
-
+    tagTable.deleteRow(tagRow.rowIndex);
 };
 
 /*
     TAG Radio
 */
-function tagRowChangeActive(tagRowCheck, tagRowAmount){
-    if(tagRowCheck.checked == 1){
-        totalTagSum(tagRowAmount.innerText);
+function tagRowChangeActive(tagRowId){
+    let tagIdCell = document.getElementById("tagRow_" + tagRowId + "_tagId");
+    let tagRowCheck = document.getElementById("tagRow_" + tagRowId + "_checkBox");
+
+    let tagDataIterator;
+    let tagData;
+    for(tagDataIterator=0; tagDataIterator < mainUnitData.tags.length; tagDataIterator++){
```

Comment:
While this type of loop does work, if you're just iterating over values in a list (not indices) you can just use
`for(let iterator of list){`
which will save you some typing and debugging.

Reply:
@EamonnMR , addressed